Feature/odm2 dev 112 114 formal expression method def

lexjansen opened the pull request.

JozefAerts reviewed (commented):
I think "Description" still needs to be added to the XML-Schema as first child element to ODMComplexTypeDefinition-MetaDataVersion.
From the documentation, I derive that SDSVarName has been by <Alias Context="SDSVarName" .../>. SDSVarName is however still in the XML-Schema. But maybe the spec is wrong...
Discrepancy between spec and schema for RangeCheck: spec provides ItemOID attribute, schema not. Which one is right?
I think ImputationMethodAttributeDefinition can be removed from the schema. It is not used anymore. However, documentation still lists it as "Deprecated" in the alphabetical list. The element definition page has been removed though.
I corrected the spec for "ReturnValue". It said that it has child elements "Parameter" and "ReturnValue". Can of course not be correct ...

JozefAerts commented:
Hi Lex,
  
  
   Excellent piece of work ... !
   
   I only found a few minor issues, please find them in the review comments on Github.
   
   After having done a "diff" for all the changes, I compared the new version of the schema with the spec on the wiki. This also allowed me to make some corrections to the wiki-spec, mostly typos, adding links, etc.. 
   
  
  
   
  
  
   I also see that with the introduction of "Code" as child element of "FormalExpression", I need to update some of the examples ...
   
  
  
   
  
  
   With best regards,
   Jozef
   
  
  
   
    Lex Jansen ***@***.***> hat am 08.04.2022 22:30 geschrieben:
   
   
    
   
   
    
   
   
   You can view, comment on, or merge this pull request online at:
     https://github.com/cdisc-org/DataExchange-ODM/pull/14
   Commit Summary
   
    4633dca Add FormalExpression Code and ExternalCodeLib
    3f84ece Add MethodSignature to MethodDef
   
   File Changes
   (3 files)
   
    M schema/ODM-foundation.xsd (21)
    M schema/ODM-study.xsd (135)
    A test/ODM2DEF_112-114-methoddef.xml (61)
   
   Patch Links:
   
    https://github.com/cdisc-org/DataExchange-ODM/pull/14.patch
    https://github.com/cdisc-org/DataExchange-ODM/pull/14.diff
   
   —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because your review was requested.Message ID: ***@***.***>

lexjansen commented:
Thanks, Jozef!

- I added MetaDataVersion/Description
- I removed ItemDef/@SDSVarName
- I removed ImputationMethodAttributeDefinition
- Changed AuditRecord/@UsedImputationMethod to UsedMethod
- Removed AuditRecord/@ID (deprecated)
- Removed ItemDef/ExternalQuestion (deprecated)
- I see examples of RangeCheck/@ItemOID, so I added to schema
